sdk/src/main/java/io/opentelemetry/sdk/distributedcontext/DistributedContextSdk.java

```diff
@@ -33,31 +34,31 @@
 import javax.annotation.concurrent.Immutable;
 
 @Immutable
-// TODO: Migrate to AutoValue
-// @AutoValue
-class DistributedContextSdk implements DistributedContext {
+@AutoValue
+abstract class DistributedContextSdk implements DistributedContext {
+
+  DistributedContextSdk() {}
 
   // The types of the EntryKey and Entry must match for each entry.
-  private final Map<EntryKey, Entry> entries;
-  @Nullable private final DistributedContext parent;
+  abstract Map<EntryKey, Entry> getEntries();
+
+  @Nullable
+  abstract DistributedContext getParent();
 
   /**
-   * Creates a new {@link DistributedContextSdk} with the given entries.
+   * Returns a new {@link Builder}.
    *
-   * @param entries the initial entries for this {@code DistributedContextSdk}.
-   * @param parent providing a default set of entries
+   * @return a {@code Builder}.
    */
-  private DistributedContextSdk(
-      Map<? extends EntryKey, ? extends Entry> entries, DistributedContext parent) {
-    this.entries = Collections.unmodifiableMap(new HashMap<>(checkNotNull(entries, "entries")));
-    this.parent = parent;
+  static Builder builder() {
+    return new AutoValue_DistributedContextSdk.Builder().setEntries(new HashMap<EntryKey, Entry>());
```

Comment:
Don't we need here to pass this as an unmodifiable `Map` as well?

Reply:
Nope, because `Builder` is mutable and should allow new entries to be set to the map.

Reply:
Oh, nevermind - saw this is the `Builder`, not the `DefaultContext` object itself ;)